module/zfs/dmu_send.c

```diff
@@ -2150,6 +2151,7 @@ struct receive_writer_arg {
 	boolean_t resumable;
 	boolean_t raw;
 	uint64_t last_object, last_offset;
+	uint64_t last_touched_object;
```

Comment:
I think using `last_touched_object` for a name is misleading.  To me it implies the last modified object in the stream, when in fact you're tracking the `largest_object` in the stream.

Reply:
yes, I was thinking of "last" as in "highest ID", not "last" as in "last in stream". largest is also misleading though, as it could refer to size? maybe something like max_objid ?